src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1593,5 +1598,45 @@ private static void VerifyDerivedTypeConstraints(IEdmModel model, IEdmVocabulary
 
             throw new ODataException(Strings.PathParser_TypeCastOnlyAllowedInDerivedTypeConstraint(fullTypeName, kind, name));
         }
+
+        private static string ResolveEscapeFunction(string identifier, IEdmType bindingType, IEdmModel model, out string parenthesisExpression)
+        {
+            Debug.Assert(identifier != null && identifier.Length >= 1 && identifier[0] == ':');
+            Debug.Assert(bindingType != null);
+
+            IEnumerable<IEdmOperation> operations = model.FindBoundOperations(bindingType);
+            IEdmFunction function = operations.OfType<IEdmFunction>().FirstOrDefault(f => IsUrlEscapeFunction(model, f));
+            if (function == null)
+            {
+                throw ExceptionUtil.CreateBadRequestError(ODataErrorStrings.RequestUriProcessor_NoBoundEscapeFunctionSupported(bindingType.FullTypeName()));
+            }
+
+            if (function.Parameters == null || function.Parameters.Count() != 2 || !function.Parameters.ElementAt(1).Type.IsString())
+            {
+                throw ExceptionUtil.CreateBadRequestError(ODataErrorStrings.RequestUriProcessor_EscapeFunctionMustHaveOneStringParameter(function.FullName()));
+            }
+
+            parenthesisExpression = function.Parameters.ElementAt(1).Name + "='" + identifier.Substring(1) + "'";
+            return function.FullName();
+        }
+
+        internal static bool IsUrlEscapeFunction(IEdmModel model, IEdmFunction function)
+        {
+            Debug.Assert(model != null);
+            Debug.Assert(function != null);
+
+            IEdmVocabularyAnnotation annotation = model.FindVocabularyAnnotations<IEdmVocabularyAnnotation>(function,
+                Edm.Vocabularies.Community.V1.CommunityVocabularyModel.UrlEscapeFunctionTerm).FirstOrDefault();
+            if (annotation != null)
+            {
+                IEdmBooleanConstantExpression tagConstant = annotation.Value as IEdmBooleanConstantExpression;
+                if (tagConstant != null)
+                {
+                    return tagConstant.Value;
+                }

```

Comment:
if tagConstant == null, shouldn't we return true?

Reply:
I think we should limit the annotation only boolean type, right. otherwise, we should ignore the annotation because maybe it's other meaning.

---
In reply to: [255711329](https://github.com/OData/odata.net/pull/1400#discussion_r255711329) [](ancestors = 255711329)

Reply:
For tag constants, if the annotation is applied but a value is not specified then the value is assumed to be true.

---
In reply to: [256082113](https://github.com/OData/odata.net/pull/1400#discussion_r256082113) [](ancestors = 256082113,255711329)

Reply:
Ok, Thanks.

---
In reply to: [267886718](https://github.com/OData/odata.net/pull/1400#discussion_r267886718) [](ancestors = 267886718,256082113,255711329)

Reply:
To handle null value == true, I think you want return true; on line 1645.

---
In reply to: [268326230](https://github.com/OData/odata.net/pull/1400#discussion_r268326230) [](ancestors = 268326230,267886718,256082113,255711329)